homeassistant/components/omnilogic/__init__.py

```diff
@@ -0,0 +1,217 @@
+            update_interval=timedelta(seconds=polling_interval),
+        )
+
+    def update_listeners(self):
+        """Call update on all listeners."""
+        for update_callback in self._listeners:
+            update_callback()
+
+    async def _async_update_data(self):
+        """Fetch data from OmniLogic."""
+        try:
+            _LOGGER.debug("Updating the coordinator data.")
+            data = await self.api.get_telemetry_data()
+            return data
+
+        except OmniLogicException as error:
+            raise UpdateFailed(f"Error updating from OmniLogic: {error}") from error
+
+
+class OmniLogicEntity(CoordinatorEntity):
+    """Defines the base OmniLogic entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        icon: str,
+        sensordata: dict,
+    ):
+        """Initialize the OmniLogic Entity."""
+        super().__init__(coordinator)
+
+        if bow != {}:
+            sensorname = (
+                backyard["BackyardName"].replace(" ", "_")
+                + "_"
+                + bow["Name"].replace(" ", "_")
+                + "_"
+                + kind
+            )
+        else:
+            sensorname = backyard["BackyardName"].replace(" ", "_") + "_" + kind
+
+        self._kind = kind
+        self._name = None
+        self.entity_id = ENTITY_ID_FORMAT.format(sensorname)
+        self._unique_id = ENTITY_ID_FORMAT.format(sensorname)
+        self._backyard = backyard
+        self._backyard_name = backyard["BackyardName"]
+        self._state = None
+        self._icon = icon
+        self._bow = bow
+        self.coordinator = coordinator
+        self.bow = bow
+        self.sensordata = sensordata
+        self._attrs = {"MspSystemId": backyard["systemId"]}
+        self.alarms = []
+        self._unsub_dispatcher = None
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the entity."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self):
+        """Return the attributes."""
+        return self._attrs
+
+    @property
+    def device_info(self):
+        """Define the device as back yard/MSP System."""
+
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self._attrs["MspSystemId"])},
+            ATTR_NAME: self._backyard.get("BackyardName"),
+            ATTR_MANUFACTURER: "Hayward",
+            ATTR_MODEL: "OmniLogic",
+        }
+
+    async def async_update(self):
+        """Update Omnilogic entity."""
+        await self.coordinator.async_request_refresh()
+
+    async def async_added_to_hass(self):
```

Comment:
I believe this is the identical to the implementation in `CoordinatorEntity` so not needed

Reply:
Removed